README.md

```diff
@@ -1,11 +1,47 @@
-# ECC Python Cookiecutter Template
+# Python Cookiecutter Template
 
 This is a Cookiecutter template for creating a Python project with Poetry,
 including configurations for various project settings.
 
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Installation](#installation)
+- [Usage](#usage)
+- [Configurations](#configurations)
+  - [Package manager - Poetry](#package-manager---poetry)
+  - [Linters and formatters - Ruff](#linters-and-formatters---ruff)
+  - [Spell checker - Typos](#spell-checker---typos)
+  - [Static type checker - Mypy](#static-type-checker---mypy)
+  - [Documentation - Sphinx](#documentation---sphinx)
+  - [Testing - pytest and pytest-cov](#testing---pytest-and-pytest-cov)
+- [CI/CD - GitHub Actions](#cicd---github-actions)
+- [Disclaimer](#disclaimer)
+- [Contributing](#contributing)
+- [Code of Conduct](#code-of-conduct)
+- [Versioning](#versioning)
+- [License](#license)
+- [Contact](#contact)
+
 ## Usage
 
-1. **Install Cookiecutter** (if you haven't already):
+1. **Install cruft** (if you haven't already):
+
+```sh
+pip install cruft
+```
+
+2. **Generate a New Project**:
+
+```sh
+cruft create https://github.com/elixir-cloud-aai/cookiecutter-python.git
+```
+
```

Comment:
Maybe add a third step:

> Commit and push the generated files as you see fit, as applicable to your Git workflow.

And add a note with `>`:

> Note that if you want to push your code to GitHub and make use of the generated GitHub Actions workflows, we recommend you to [set up the necessary secrets](#ADD-LINK-TO-RELEVANT-SECTION) before pushing the code. Otherwise some of the workflows may fail.